pgsync/sync.py

```diff
@@ -280,39 +280,20 @@ def logical_slot_changes(
                 continue
             _rows.append(row)
 
+        # We now just dump this in the redis queue and let it process in the same manner
+        # as the "realtime" changes.  If it's a one-time load, we don't care about what's next. This
+        # has the advantage of letting the redis sorted-sets reduce a lot of the duplicate work.
         for i, row in enumerate(_rows):
-
             logger.debug(f"txid: {row.xid}")
             logger.debug(f"data: {row.data}")
-            # TODO: optimize this so we are not parsing the same row twice
             try:
                 payload = self.parse_logical_slot(row.data)
+                self.redis.push(payload, row.xid)
```

Comment:
*** This is really the heart of the change.  There is some reformatting in here.